src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationReloadToken.cs

```diff
@@ -33,11 +33,60 @@ public class ConfigurationReloadToken : IChangeToken
         /// <param name="callback">The callback to invoke.</param>
         /// <param name="state">State to be passed into the callback.</param>
         /// <returns>The <see cref="CancellationToken"/> registration.</returns>
-        public IDisposable RegisterChangeCallback(Action<object?> callback, object? state) => _cts.Token.Register(callback, state);
+        public IDisposable RegisterChangeCallback(Action<object?> callback, object? state)
+        {
+#if NETCOREAPP || NETSTANDARD2_1
+            try
+            {
+                return _cts.Token.UnsafeRegister(callback, state);
```

Comment:
>UnsafeRegister [](http://example.com/codeflow?start=34&length=14)

I am not fully getting this. The API supposes to do register the `callback`. right? What exactly the behavior when running on NetCoreApp?

Reply:
Both Register and UnsafeRegister register the callback.  The sole difference is the former captures the current ExecutionContext and uses it if/when the callback is invoked, and the latter does not.